### DISCUSSION TITLE
Quicksearch functionality returns 'communication failure'

### DISCUSSION THREAD

**Discussion by oriiyx** [pimcore/pimcore#8803] [Q&A / Support]
`Status: 0 | communication failure
URL: /admin/search/search/quicksearch?_dc=1618398006506
Method: GET
Message: 
undefined`

### Okay - this is going to be a longer post - since we've already researched the issue - but we've ran out of ideas to test and debug.

First I would like to specify that this is an isolated issue - we have 5 different Pimcore projects that don't have this issue - so it has to do with server/database/codebase of this specific project.

#### Pimcore version is "v6.6.4".

So when our customer uses the quicksearch function (ctrl+shift+F or MacOS cmd+shift+F) and inputs the SKU (or EAN) number in the search box - it returns the error that I copy pasted above. The error doesn't happen all the time - sometimes it works great - sometimes you get the error on every try. It is pretty inconsistent and looking at htop of the server - the resources don't really show any signs of overloading the server when using the search.
If we don't get a return of answer in 30 seconds the XHRequest fails and exception is rendered to the frontend. (Main error we're trying to fix).

For everyone that doesn't know how the quicksearch function works - here's a fast one:
when you type into the search box - once you have 4 characters or more - on each "keyup" javascript will send a request to "/admin/search/search/quicksearch" controller with GET method - param will contain the search input you typed in.
The Controller picks up the quicksearchAction inside the SearchController - and the query that returns the items is this one:

` 1784 Query	SELECT * FROM search_backend_data WHERE (( MATCH (`data`,`properties`) AGAINST ('1234*' IN BOOLEAN MODE) ) AND type != 'folder')   ORDER BY ( MATCH (`data`,`properties`) AGAINST ('1234*' IN BOOLEAN MODE) ) DESC LIMIT 50
`
(When I tried to run the upper query with random parameters that were not cached I got normal speeds:

- 28 rows in set (0.021 sec)
- Empty set (0.003 sec)
- Empty set (0.002 sec)
- Empty set (0.005 sec)

All the items are then processed and prepared with correct information and send to render in view to the frontend.

#### Our first go-to is to clear the cache.

Since it didn't work - we start to investigate further.
We have 3 different environments - local - dev and production - and the issue can be replicated in all 3 of them.

We spoke with our server maintainer and he said to increase RAM for another 8GB - that it could be the issue of running out of memory.

We noticed that the issue still remained.

#### I tried to run "pimcore:search-backend-reindex" command and waited for process to reindex.
Still no luck!

I then tried to degrade my local environment to try to see if different Pimcore version would help - I tried:

- 6.4
- 6.5

Then I thought that mby they found the bug and fixed it in 6.7 or 6.8 - but the bug still remained.

So I reverted back to my original composer.lock file settings for the project.

I checked Symfony profiler of the requests - and it shows that 90% of the time was used by the controller.

So I went to debugging mysql next:
#### I enabled slow query monitoring and set to log all the queries that are longer than 1 second - it didn't record a single record.
Then I enabled general query monitoring and tailed the file while I proceeded to replicate the bug
#### What I noticed was that the queries were coming in with a delay - the more I wrote numbers in the search field - the more it was getting delayed.
But since the bug can be replicated with a single copy paste of SKU into the search field - I can't confirm this as a cause.

The codebase - we use this pimcore project as a simple stop in between a ERM and Magento shop - primary function is importing of objects (mainly products).
We don't have much of logic and I went through all the controllers and events that are there. Even from Symfony profiler I would be able to see if anything else was being processed.

If anyone has any idea what else to test or research - I'd be more than glad to try!
Thank in advance to anyone who has the time to chip in the help!

**Comment by GaneshKandu** [comment]
```
Timestamp: Thu Jul 21 2022 16:34:18 GMT+0530 (India Standard Time)
Status: 0 | communication failure
URL: /admin/targeting/target-group/list?_dc=1658401428218
Method: GET
Message: 
undefined
```

I am getting same issue

did you resolved this error ?

**Reply by oriiyx** [reply]
@GaneshKandu 

Hey mate,

we fixed it by repairing corrupted tables in our database.
Hope it solves your issues!

Cheers,

**Reply by mohit-kumar1** [reply]
@GaneshKandu 
Did you get any solution of it.

**Reply by GaneshKandu** [reply]
No

**Comment by krishanumandalnavsoft** [comment]
```
Timestamp: Mon Nov 07 2022 18:45:31 GMT+0530 (India Standard Time)
Status: 0 | communication failure
URL: /admin/object/grid-proxy?classId=2&folderId=244&xaction=read&_dc=1667826901574
Method: GET
Message: 
undefined
```
Hello, All I'm also facing same problem on pimcore 6.9. Please help me.

**Comment by ronj-soft** [comment]
any update on this issue?
